migration of login component (home, navbar, activate, reset finish, register)

flaviencathala opened the pull request.

wmarques commented on generators/client-2/templates/src/main/webapp/app/account/_account.module.ts:
```diff
@@ -35,7 +35,7 @@ import { SettingsComponent } from "./settings/settings.component";
 <% if (enableTranslation) { %>upgradeAdapter.upgradeNg1Provider('$translate');<% } %>
 upgradeAdapter.upgradeNg1Provider('Auth');
-upgradeAdapter.upgradeNg1Provider('LoginService');
+//upgradeAdapter.upgradeNg1Provider('LoginService');
```
I think you can remove this line.

wmarques commented on generators/client-2/templates/src/main/webapp/app/components/_common.module.ts:
```diff
@@ -15,7 +15,7 @@ import { AuthServerProvider } from './auth/auth-jwt.service';
 <%_ } _%>
 import { Account } from './auth/account.service';
-import { LoginService } from './login/login.service';
+//import { LoginService } from './login/login.service';
```
same, remove commented lines in this file as we import the login service in appmodule now.

wmarques commented:
@flaviencathala You have conflicts, please rebase your branch to the current `angular-2` branch.

flaviencathala commented:
Done !

ruddell commented:
Didn't test it but the changes look good.  For Travis, you need to fix `test-creation.js` to check for the component instead of the controller.

wmarques commented:
@flaviencathala Fix test and cleanup a bit the code by removing useless comments and I'll merge this :)

deepu105 commented on generators/client-2/templates/src/main/webapp/app/account/_account.ng2module.ts:
```diff
@@ -1,6 +1,7 @@
 import { NgModule } from '@angular/core';
 import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
 import { UIRouterModule } from "ui-router-ng2";
+import { BrowserModule } from '@angular/platform-browser';
```
why is this added? its already added in the root module

deepu105 commented on generators/client-2/templates/src/main/webapp/app/account/_account.ng2module.ts:
```diff
@@ -46,10 +47,12 @@ let ACCOUNT_STATES = [
         UIRouterModule.forChild({ states: ACCOUNT_STATES })
     ],
     declarations: [
+        //<%=jhiPrefixCapitalized%>LoginModalComponent,
```
this shouldn't be imported here it should be imported in the common module and exported from there so that it can be used in all other modules

deepu105 commented on generators/client-2/templates/src/main/webapp/app/_app.ng2module.ts:
```diff
@@ -26,12 +27,12 @@ import { ActiveMenuDirective } from './layouts/navbar/active-menu.directive';
         <%=angular2AppName%>AccountModule
     ],
     declarations: [
+        <%=jhiPrefixCapitalized%>LoginModalComponent,
```
this shouldn't be imported here it should be imported in the common module and exported from there so that it can be used in all other modules

flaviencathala replied:
Do you mean export it from app module and then import it in account module from the app module ? What about home and navbar components do i import it directly in the components ?

deepu105 replied:
no just declare `LoginModalComponent` and then export it from the `common.ng2module` as below, now import the `<%=angular2AppName%>CommonModule` in the `account.ng2module` and you should be able to access any exported members from there. home and navbar are not reused hence they can be in the `app.ng2module`
```
@NgModule({
    imports: [
        <%=angular2AppName%>SharedModule
    ],
    declarations: [
        jhiAlertComponent,
        PageRibbonComponent,
        <%=jhiPrefixCapitalized%>LoginModalComponent
    ],
    providers: [
        AlertService,
        ProfileService,<%_ if (enableTranslation){ _%>
        <%=jhiPrefixCapitalized%>LanguageService,
        <%_ } _%>
        Account,
        Principal,
        AuthService
    ],
    exports: [
        <%=jhiPrefixCapitalized%>LoginModalComponent
    ],
    schemas: [CUSTOM_ELEMENTS_SCHEMA]

})
export class <%=angular2AppName%>CommonModule {}
```

flaviencathala replied:
It works perfectly, thanks !

deepu105 commented on generators/client-2/templates/src/main/webapp/app/account/_account.ng2module.ts:
```diff
@@ -46,10 +47,12 @@ let ACCOUNT_STATES = [
 
 @NgModule({
     imports: [
+        BrowserModule,
```
plz remove this its not required here

deepu105 commented on generators/client-2/templates/src/main/webapp/app/account/activate/_activate.component.ts:
```diff
@@ -1,16 +1,22 @@
+    constructor(@Inject('Auth') Auth,
+                private modalService: NgbModal,
+                @Inject('$stateParams') $stateParams) {
+        this.Auth = Auth;
```
not required just add private to the argument declaration in the constructor

deepu105 commented on generators/client-2/templates/src/main/webapp/app/account/activate/_activate.component.ts:
```diff
@@ -23,4 +29,14 @@ export class ActivateComponent implements OnInit {
     }
 
+    login(template) {
+        console.log('login activate');
```
remove this console log

deepu105 commented on generators/client-2/templates/src/main/webapp/app/account/password-reset/finish/_password-reset-finish.component.ts:
```diff
@@ -14,14 +14,19 @@ export class PasswordResetFinishComponent implements OnInit {
+                @Inject('$stateParams') $stateParams,
+                private elementRef: ElementRef,
+                private renderer: Renderer) {
+        this.Auth = Auth;
```
same here use private arguments

deepu105 commented on generators/client-2/templates/src/main/webapp/app/account/password-reset/finish/_password-reset-finish.component.ts:
```diff
@@ -14,14 +14,19 @@ export class PasswordResetFinishComponent implements OnInit {
 
     ngOnInit() {
         this.resetAccount = {};
-        this.login = this.LoginService.open;
```
how is this handled

deepu105 commented on generators/client-2/templates/src/main/webapp/app/components/_common.ng2module.ts:
```diff
@@ -10,9 +11,11 @@ import { jhiAlertComponent } from "./alert/alert.component";
 
 @NgModule({
     imports: [
+        BrowserModule,
```
why?

deepu105 commented on generators/client-2/templates/src/main/webapp/app/components/login/_login.component.ts:
```diff
@@ -0,0 +1,81 @@
+    }
+
+    login () {
+        let vm = this;
```
not required use arrow functions to access `this` directly

deepu105 commented on generators/client-2/templates/src/main/webapp/app/components/login/_login.component.ts:
```diff
@@ -0,0 +1,81 @@
+                vm.$state.go('home');
+            }
+
+            //vm.$rootScope.$broadcast('authenticationSuccess');
```
we need to emit an observable here and subscribe that event

flaviencathala replied:
It throws me an error ("No provider for $rootScope!") so i commented it the time i found a solution. Did you have some problems when injecting $rootScope in the alert-error module ? Thanks.

deepu105 commented on generators/client-2/templates/src/main/webapp/app/components/login/login.html:
```diff
@@ -1,6 +1,8 @@
-            ng-click="vm.cancel()">&times;</button>
+    <button aria-label="Close" data-dismiss="modal" class="close" type="button" (click)="dismiss('closed')"><span aria-hidden="true">x</span>
+    </button>
+    <!--<button type="button" class="close" data-dismiss="modal" aria-hidden="true"
```
remove

flaviencathala replied:
If i remove it the only way to close the window will be to click outside of the modal. I think the cross on the top right corner is very useful.

deepu105 replied:
I was talking about the commented out code, remove commented code

flaviencathala replied:
Ah ok

deepu105 commented on generators/client-2/templates/src/main/webapp/app/components/login/login.html:
```diff
@@ -9,51 +11,37 @@ <h4 class="modal-title" translate="login.title">Sign in</h4>
                 <strong>Failed to sign in!</strong> Please check your credentials and try again.
             </div>
         </div>
-        <%_ if (enableSocialSignIn) { _%>
```
this shouldnt be removed

deepu105 commented on generators/client-2/templates/src/main/webapp/app/components/login/login.html:
```diff
@@ -9,51 +11,37 @@ <h4 class="modal-title" translate="login.title">Sign in</h4>
+                You don't have an account yet? <a class="alert-link" href="" (click)="register()">Register a new account</a>
             </div>
         </div>
-        <%_ if (enableSocialSignIn) { _%>
```
shouldnt be removed

deepu105 commented on generators/client-2/templates/src/main/webapp/app/layouts/navbar/navbar.html:
```diff
@@ -72,8 +72,8 @@
                         </li>
-                        <li uiSrefActive="active" *ngSwitchCase="false">
-                            <a href="" (click)="login()" id="login">
+                        <li ui-sref-active="active" *ngSwitchCase="false">
```
remove `ui-sref-active="active"` it not required

deepu105 commented on generators/client-2/templates/src/main/webapp/app/account/activate/_activate.component.ts:
```diff
@@ -23,4 +29,14 @@ export class ActivateComponent implements OnInit {
 
+    login(template) {
+        console.log('login activate');
+        this.modalRef = this.modalService.open(template);
```
we need a better way for this. lets just use the loginService like before and do the result handling within that and return the modelRef from there

deepu105 commented on generators/client-2/templates/src/main/webapp/app/account/_account.module.ts:
```diff
@@ -39,6 +40,7 @@ angular
         'ui.bootstrap',
         'ui.router'
     ])
+    .directive('<%=jhiPrefix%>LoginModal', <angular.IDirectiveFactory> upgradeAdapter.downgradeNg2Component(<%=jhiPrefixCapitalized%>LoginModalComponent))
```
I dont think its required here. If needed do it in the common module

deepu105 commented:
please also rebase against latest branch as there are lot of changes in auth services

flaviencathala commented:
@deepu105 
I added the result handling in the loginService, did all the small modifications you asked and rebased.
It works for home screen, navbar, activate and reset password screens.
It fails on register screen, and the injection of $rootScope in login component still throws me an error.
I'm on it.

flaviencathala commented:
@deepu105 The $rootScope error has been fixed with the rebase, could you please merge with angular-2 if it's ok for you ? Thanks.
